Lombiq.Tests.UI/Services/OrchardCoreHosting/OrchardApplicationFactory.cs

```diff
@@ -42,7 +47,14 @@ public sealed class OrchardApplicationFactory<TStartup> : WebApplicationFactory<
     protected override IHost CreateHost(IHostBuilder builder)
     {
         builder.ConfigureHostConfiguration(configurationBuilder => _configureHost?.Invoke(configurationBuilder));
-        return base.CreateHost(builder);
+        // This lock is to avoid parallel start of the application. The
+        // Microsoft.Extensions.Hosting.HostFactoryResolver.HostingListener.CreateHost() starts a new thread for the
+        // web application instance which can cause issues in eg.: NLog.Config.Factory<TBaseType, TAttributeType>.RegisterDefinition()
+        // which is using not thread safe Dictionary to store cached types when initializing default logger instance.
```

Comment:
```suggestion
        // This lock is to avoid parallel start of the application.
        // Microsoft.Extensions.Hosting.HostFactoryResolver.HostingListener.CreateHost() starts a new thread for the web
        // application instance which can cause issues in e.g.:
        // NLog.Config.Factory<TBaseType, TAttributeType>.RegisterDefinition() which is using non-thread-safe Dictionary
        // to store cached types when initializing the default logger instance.
```